modules/dnn_superres/src/dnn_superres_quality.cpp

```diff
@@ -0,0 +1,470 @@
+            split(orig, orig_channels);
+
+            Mat img_float;
+            img_channels[0].convertTo(img_float, CV_32F);
+
+            Mat orig_float;
+            orig_channels[0].convertTo(orig_float, CV_32F);
+
+            Mat mu_img, mu_orig;
+            GaussianBlur( img_float, mu_img, Size(11, 11), 0, 0 );
+            GaussianBlur( orig_float, mu_orig, Size(11, 11), 0, 0 );
+            Mat mu_mul = mu_img.mul(mu_orig);
+            mu_img = mu_img.mul(mu_img);
+            mu_orig = mu_orig.mul(mu_orig);
+
+            Mat sigma_img, sigma_orig;
+            GaussianBlur( img_float.mul(img_float), sigma_img, Size(11, 11), 0, 0 );
+            GaussianBlur( orig_float.mul(orig_float), sigma_orig, Size(11, 11), 0, 0 );
+            sigma_img = sigma_img - mu_img;
+            sigma_orig = sigma_orig - mu_orig;
+
+            Mat sigma_mul;
+            GaussianBlur( img_float.mul(orig_float), sigma_mul, Size(11, 11), 0, 0 );
+            sigma_mul = sigma_mul - mu_mul;
+
+            double c1 = (0.01 * 255) * (0.01 * 255);
+            double c2 = (0.03 * 255) * (0.03 * 255);
+
+            Mat nom1 = (2 * mu_mul + c1);
+            Mat nom2 = (2 * sigma_mul + c2);
+
+            Mat nominator = nom1.mul(nom2);
+
+            Mat denom1 = mu_img + mu_orig + c1;
+            Mat denom2 = sigma_img + sigma_orig + c2;
+
+            Mat denominator = denom1.mul(denom2);
+
+            Mat ssim = nominator / denominator;
+
+            Scalar mssim = mean(ssim);
+
+            return mssim[0];
+        }
+
+        void DnnSuperResQuality::benchmark(DnnSuperResImpl sr, Mat img,
+                                            std::vector<double>& psnrValues,
+                                            std::vector<double>& ssimValues,
+                                            std::vector<double>& perfValues,
+                                            bool showImg,
+                                            bool showOutput)
+        {
+            CV_Assert(!img.empty());
+
+            psnrValues = std::vector<double>();
+            ssimValues = std::vector<double>();
+            perfValues = std::vector<double>();
+
+            if(showOutput)
+                std::cout << "Start benchmarking" << std::endl;
+
+            int scale = sr.getScale();
+
+            int width = img.cols - (img.cols % scale);
+            int height = img.rows - (img.rows % scale);
+            Mat cropped = img(Rect(0, 0, width, height));
+            Mat imgDownscaled;
+
+            resize(cropped, imgDownscaled, Size(), 1.0/scale, 1.0/scale);
+
+            Mat imgUpscaled;
+
+			double elapsed = 0.0;
+
+            #ifdef _WIN32
+                LARGE_INTEGER st, et, el, fq;
+            #else
+                double start, end;
+                timespec ts_beg, ts_end;
+            #endif
+
+            #ifdef _WIN32
+                QueryPerformanceFrequency(&fq);
+                QueryPerformanceCounter(&st);
```

Comment:
Use `int64 getTickCount();` or `TickMeter` helper class instead:
https://github.com/opencv/opencv/blob/4.1.1/modules/core/include/opencv2/core/utility.hpp#L285-L332

Reply:
Changed it to TickMeter.